src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java

```diff
@@ -515,6 +518,10 @@ public void setupMetadata(FeatureTypeInfo ftinfo, FeatureSource featureSource)
     public ReferencedEnvelope getLatLonBounds(ReferencedEnvelope nativeBounds,
             CoordinateReferenceSystem declaredCRS) throws IOException {
         if (nativeBounds != null && declaredCRS != null) {
+            // shortcut: return empty envelope if nativeBounds is empty
+            if (nativeBounds.isEmpty()) {
+                return new ReferencedEnvelope(nativeBounds, DefaultGeographicCRS.WGS84);
```

Comment:
Well, it is XML valid, not sure if it is spec compliant though... it just felt weird to see a null envelope (i.e. minx > maxx, miny > maxy) being transformed to a non-null one, as the previous code was doing.

I can revert this change, if you think it is not correct.